ldap/ldap.go

```diff
@@ -230,8 +244,8 @@ func (u *userService) bind(username, password string) error {
 // search wraps u.conn.Search, reconnecting if the LDAP server has closed the connection.
 // https://github.com/go-ldap/ldap/issues/113
 func (u *userService) search(searchRequest *ldap.SearchRequest) (*ldap.SearchResult, error) {
-	if err := u.ensureConnected(); err != nil {
-		return nil, errors.Wrap(err, "problem connecting to ldap server")
+	if err := u.login(u.serviceUserName, u.serviceUserPassword); err != nil {
```

Comment:
Does this need its own service path as well? I'm not sure if this service user will have the same path as the service users that we currently have.

Reply:
I don't know, but I added one specifically for the service user.